WordPress/Classes/ViewRelated/Notifications/Comment Moderation/CommentModerationSheetHostingView.swift

```diff
@@ -0,0 +1,75 @@
+import UIKit
+import SwiftUI
+
+final class CommentModerationSheetHostingView: UIView {
```

Comment:
This is the most important file in this PR and also the view that resolves the bug.

For context, the [bug](https://github.com/wordpress-mobile/WordPress-iOS/pull/23225#issuecomment-2122713266) was occurring because the moderation hosting view was not resizing automatically when its content changed. This is a known issue in UIKit and SwiftUI integration, where the hosting view doesn't resize when the SwiftUI view's size changes.

While it's possible to resize the hosting view by calling `hostingView.invalidateIntrinsicContentSize`, the resize will occur without animation which might break the moderation view animation.


There is a workaround to resolve this issue and it works as following:

1. Constrain the moderation hosting view to the edges of `viewController.view` so that it matches the size of its parent view.
2. By default, the Hosting View will prevent touches from passing through. To allow touches outside the moderation view to pass through, we are overriding the hosting view `hitTest` method. 
2. Make the Hosting View transparent.
3. Now the Bottom Sheet can resize freely, and the user doesn't see the “Hosting View”.